galite-core/src/main/kotlin/org/kopi/galite/report/VReport.kt

```diff
@@ -15,6 +15,641 @@
+      setCommandEnabled(cmdColumnInfo!!, column != -1)
+    }
+    if (cmdEditColumn != null) {
+      setCommandEnabled(cmdEditColumn!!, column != -1 && model.getAccessibleColumn(column).addedAtRuntime)
+    }
+  }
+  // ----------------------------------------------------------------------
+  // IMPLEMENTATION
+  // ----------------------------------------------------------------------
+  /**
+   * Returns the selected column or -1 if no column is selected.
+   */
+  private val selectedColumn: Int
+    private get() = (getDisplay() as UReport).selectedColumn
+
+  /**
+   * Returns the selected cell or !!! ??? if no cell is selected.
+   */
+  private val selectedCell: Point
+    private get() = (getDisplay() as UReport).selectedCell
+
+  fun genHelp(): String? {
+    val fileName: String
+    val surl = StringBuffer()
+    fileName = VHelpGenerator().helpOnReport(pageTitle,
+            commands,
+            model,
+            help)
+    return if (fileName == null) {
+      null
+    } else {
+      try {
+        surl.append(File(fileName).toURL().toString())
+      } catch (mue: MalformedURLException) {
+        throw InconsistencyException(mue)
+      }
+      surl.toString()
+    }
+  }
+
+  fun showHelp() {
+    VHelpViewer().showHelp(genHelp())
+  }
+
+  // ----------------------------------------------------------------------
+  // DATA MEMBERS
+  // ----------------------------------------------------------------------
+  private var cmdFold: VCommand? = null
+  private var cmdUnfold: VCommand? = null
+  private var cmdSort: VCommand? = null
+  private var cmdOpenLine: VCommand? = null
+  private var cmdFoldColumn: VCommand? = null
+  private var cmdUnfoldColumn: VCommand? = null
+  private var cmdColumnInfo: VCommand? = null
+  private var cmdEditColumn: VCommand? = null
+
+  /**
+   * Set the source for this document
+   */
+  private lateinit var source: String
+  override fun getSource(): String? = null
+
+  protected var model: MReport
+  private var built = false
+  private var pageTitle = ""
+  private var firstPageHeader = ""
+
+  // ----------------------------------------------------------------------
+  // HELP
+  // ----------------------------------------------------------------------
+  var help: String? = null
+  protected var VKT_Triggers: Array<IntArray>? = null
+  protected var commands: Array<VCommand>? = null
+  private val activeCommands: Vector<VCommand>
+  /**
+   * sets the print options
+   */
+  /**
+   * sets the print options
+   */
+  // print configuration object
+  var printOptions: PConfig
+  /**
+   * Get the media for this document
+   */
+  /**
+   * Set the media for this document
+   */
+  var media: String? = null
+  // ----------------------------------------------------------------------
+  // CONSTRUCTORS
+  // ----------------------------------------------------------------------
+  /**
+   * Constructor
+   */
+  /**
+   * Constructor
+   */
+  init {
+    model = MReport()
```

Comment:
we can remove this line and add declaration to : `var model: MReport = MReport()`